collect_app/build.gradle

```diff
@@ -65,6 +65,7 @@ allprojects {
     }
 }
 dependencies {
+    compile 'org.jdeferred:jdeferred-android-aar:1.2.4'
```

Comment:
I like the way that this code is structured and I see the value of promises in this context. That said, it brings up some questions about how the project will manage adding external dependencies. It seems it should be something that is carefully considered. I think it might be worth discussing in the Slack a bit.